lib/house.rb

```diff
@@ -0,0 +1,46 @@
+class House
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    @price[1..-1].to_i > 500000 ? true : false
+  end
+
+  def rooms_from_category(category)
+    @rooms.select {|room| room.category == category}
+  end
+
+  def area
+    @rooms.sum {|room| room.area}
+  end
+
+  def details
+    {"price" => @price[1..-1].to_i, "address" => @address}
+  end
+
+  def price_per_square_foot
+    (@price[1..-1].to_f / area).round(2)
+  end
+
+  def rooms_sorted_by_area
+    (@rooms.sort_by {|room| room.area}).reverse
+  end
+
+  def rooms_by_category
+    rmc = {}
+    categories = @rooms.uniq {|room| room.category}
+    categories.each do |category|
+      rmc[category] = rooms_from_category(category)
```

Comment:
nice job setting key/value pairs, and nice job using the return value of another method here. Ultimately, this doesn't work, because you're passing in a room instead of a category here, which means that every value is an empty array (since it can't find any rooms with a category that matches a room object), but if `categories` had been correct, this would have been perfect.